### PR TITLE
following symlinks when reading blocks

### PR DESCRIPTION
**Pull request by RCasatta**
In my setup I have symlinks for blocks, this PR allows to read the file content pointed by the symlink

### CODE REVIEW

File: src/blockchain/utils/blkfile.rs [hunk 1]
```diff
@@ -42,21 +42,29 @@ impl BlkFile {
...
+                let mut metadata :Metadata = de.metadata().unwrap();
+                if sl {
+                    path = fs::read_link(path.clone()).unwrap();
+                    metadata = fs::metadata(path.clone()).unwrap();
+                }
+
+                let file_name = String::from(transform!(path.as_path().file_name().unwrap().to_str()));
+
+                // Check if it's a valid blk file
+                if let Some(index) = BlkFile::parse_blk_index(&file_name, &blk_prefix, &blk_ext) {
```
Comment (gcarq) [inline src/blockchain/utils/blkfile.rs]: Is the `is_file()` check no longer necessary?, what happens when `file_name` is a directory?

**Comment by gcarq** [comment]
Hi, thanks for the PR. Please consider comments

**Comment by mikispag** [comment]
Since this is integrating my dirty "emergency fix" for Segwit block support, and likely that would be overwritten by proper support in rusty-blockparser, I wonder if it's worth adding it anyway so that the parser works now. How far is a proper fix?

**Comment by gcarq** [comment]
@mikispag Yes I think it's worth adding it, because right now I'm not that much involved into Bitcoin. I'm happy about any PR in this direction. One thought I had was if we should use some internals of parity-bitcoin for SegWit support and/or the Script parsing, so that we don't need to reinvent the wheel.

**Comment by mikispag** [comment]
Riccardo, would you mind if I send a pull request with my Segwit fix separately instead, for better tracking?

**Comment by RCasatta** [comment]
@mikispag yeah sorry that was a mistake due to lazily working on master, shame on me. Creating another proper PR

File: src/blockchain/utils/blkfile.rs (hunk 1 above)
Comment (RCasatta) [reply src/blockchain/utils/blkfile.rs]: reintroducing the `is_file()` check in another PR

**Pull request closed without merging**